starfish/pipeline/algorithmbase.py

```diff
@@ -1,6 +1,24 @@
+class AlgorithmBase:
+    """
+    This is the base class of any algorithm that starfish exposes. This base class is retrieved by
```

Comment:
this is not entirely true.  it's more like subclass of PipelineComponent retrieve subclasses of its corresponding AlgorithmBase.